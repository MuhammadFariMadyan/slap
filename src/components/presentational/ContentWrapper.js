import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import "typeface-roboto";

const Outline = styled.rect`
  fill: none;
  stroke: #686de0;
  stroke-width: 2px;
  cursor: pointer;
`;

const Point = styled.rect`
  fill: #686de0;
  stroke: white;
  stroke-width: 2px;
`;

const ContentWrapper = ({ width, height, x, y }) => {
  const pointSize = {
    width: 8,
    height: 8
  };

  const modifier = 8 / 2;

  const points = [
    {
      ...pointSize,
      x: x - modifier,
      y: y - modifier,
      style: { cursor: "nw-resize" }
    },
    {
      ...pointSize,
      x: x + width / 2 - modifier,
      y: y - modifier,
      style: { cursor: "n-resize" }
    },
    {
      ...pointSize,
      x: x + width - modifier,
      y: y - modifier,
      style: { cursor: "ne-resize" }
    },
    {
      ...pointSize,
      x: x - modifier,
      y: y + height / 2 - modifier,
      style: { cursor: "w-resize" }
    },
    {
      ...pointSize,
      x: x + width - modifier,
      y: y + height / 2 - modifier,
      style: { cursor: "e-resize" }
    },
    {
      ...pointSize,
      x: x - modifier,
      y: y + height - modifier,
      style: { cursor: "sw-resize" }
    },
    {
      ...pointSize,
      x: x + width / 2 - modifier,
      y: y + height - modifier,
      style: { cursor: "s-resize" }
    },
    {
      ...pointSize,
      x: x + width - modifier,
      y: y + height - modifier,
      style: { cursor: "se-resize" }
    }
  ];

  return (
    <g>
      <Outline width={width} height={height} x={x} y={y} />
      {points.map((point, index) => (
        <Point key={index} {...point} />
      ))}
    </g>
  );
};

ContentWrapper.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  x: PropTypes.number,
  y: PropTypes.number
};

ContentWrapper.defaultProps = {
  width: 320,
  height: 96
};

export default ContentWrapper;
