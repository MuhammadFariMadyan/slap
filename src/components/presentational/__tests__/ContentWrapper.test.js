import React from "react";
import { shallow } from "enzyme";

import ContentWrapper from "../ContentWrapper";

describe("ContentWrapper", () => {
  const setup = propOverrides => {
    const props = {
      width: 320,
      height: 48,
      x: 32,
      y: 32,
      ...propOverrides
    };

    const wrapper = shallow(<ContentWrapper {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
