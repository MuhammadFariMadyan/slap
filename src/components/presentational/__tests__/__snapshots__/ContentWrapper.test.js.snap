// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentWrapper should renders default 1`] = `
<g>
  <styled.rect
    height={48}
    width={320}
    x={32}
    y={32}
  />
  <styled.rect
    height={8}
    key="0"
    style={
      Object {
        "cursor": "nw-resize",
      }
    }
    width={8}
    x={28}
    y={28}
  />
  <styled.rect
    height={8}
    key="1"
    style={
      Object {
        "cursor": "n-resize",
      }
    }
    width={8}
    x={188}
    y={28}
  />
  <styled.rect
    height={8}
    key="2"
    style={
      Object {
        "cursor": "ne-resize",
      }
    }
    width={8}
    x={348}
    y={28}
  />
  <styled.rect
    height={8}
    key="3"
    style={
      Object {
        "cursor": "w-resize",
      }
    }
    width={8}
    x={28}
    y={52}
  />
  <styled.rect
    height={8}
    key="4"
    style={
      Object {
        "cursor": "e-resize",
      }
    }
    width={8}
    x={348}
    y={52}
  />
  <styled.rect
    height={8}
    key="5"
    style={
      Object {
        "cursor": "sw-resize",
      }
    }
    width={8}
    x={28}
    y={76}
  />
  <styled.rect
    height={8}
    key="6"
    style={
      Object {
        "cursor": "s-resize",
      }
    }
    width={8}
    x={188}
    y={76}
  />
  <styled.rect
    height={8}
    key="7"
    style={
      Object {
        "cursor": "se-resize",
      }
    }
    width={8}
    x={348}
    y={76}
  />
</g>
`;
